src/hooks/use-add-asset-to-wallet.ts

```diff
@@ -0,0 +1,60 @@
+import * as Sentry from "@sentry/react";
+import { useWeb3React } from "@web3-react/core";
+import { InjectedConnector } from "@web3-react/injected-connector";
+import React from "react";
+
+import { appEnv, Networks } from "../config";
+
+export const useAddAssetSupported = () => {
+  const { connector } = useWeb3React();
+
+  return React.useMemo(() => {
+    return (
+      connector &&
+      connector instanceof InjectedConnector &&
+      window.ethereum.isMetaMask
+    );
+  }, [connector]);
+};
+
+export const useAddAssetToWallet = (
+  address: string,
+  symbol: string,
+  decimals: number,
+  image: string
+) => {
+  const { connector } = useWeb3React();
+  const addSupported = useAddAssetSupported();
+  const add = React.useCallback(async () => {
+    try {
+      const provider = await connector?.getProvider();
+      provider.request({
+        method: "wallet_watchAsset",
+        params: {
+          type: "ERC20",
+          options: {
+            address,
+            symbol: `${symbol}${
+              // Add the environment if not mainnet
+              appEnv === Networks.MAINNET
+                ? ""
+                : // Remove NET as VEGA(TESTNET) is too long
+                  ` ${appEnv.replace("NET", "")}`
```

Comment:
Kind of feel like this should be VEGA(TEST) rather than VEGA(NET)? If it fits?

Reply:
That's what this does, we are removing the NET part so:
* stagnet -> stag
* testnet -> test
* stagnet2 -> stag2
* mainnet -> ignored